lucene/core/src/test/org/apache/lucene/search/TestIndexSearcher.java

```diff
@@ -237,4 +241,28 @@ public void testGetSlices() throws Exception {
     service.shutdown();
     IOUtils.close(r, dir);
   }
+
+  public void testOneSegmentExecutesOnTheCallerThread() throws IOException {
```

Comment:
Should we have more random testing, where we test different combination of slices? Maybe have a document distribution with random commit calls to create different segment distributions?

Reply:
RandomIndexWriter is used in this test which does that.